helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -1278,29 +1279,28 @@ private void fireDataChangedEvents(final String path, Set<IZkDataListenerEntry>
             + listener.getDataListener() + " prefetch data: " + listener.isPrefetchData()) {
           @Override
           public void run() throws Exception {
-            // Reinstall watch before listener callbacks to check the znode status
             if (!pathStatRecord.pathChecked()) {
-              pathStatRecord.recordPathStat(getStat(path, true), notificationTime);
+              pathStatRecord.recordPathStat(getStat(path, pathExists), notificationTime);
             }
-            if (!pathStatRecord.pathExists()) {
-              // no znode found at the path, trigger data deleted handler.
+            if (!pathExists) {
```

Comment:
Seems to me all the change from line 1285 is not necessary, right? We can still use the old code. 

The change may not be wrong, but seems to necessary. 
Or did I miss something here?

Reply:
Yes, we can keep the old code. I just changed the 
`if { } else {}` to `if () return; xxx` to make the code less indent. No function change

Reply:
How about let us keep the old code. The thing is that later if people looking at the change, it would take some time to figure out there is not really logical change here. 

This is more of a beautification work. This may look better, but for fix, can be confusing.

Reply:
Sure, reverted the style